pkg/csi/service/wcp/controller.go

```diff
@@ -2022,8 +2016,22 @@ func (c *controller) CreateSnapshot(ctx context.Context, req *csi.CreateSnapshot
 		var snapshotID string
 		var cnsSnapshotInfo *cnsvolume.CnsSnapshotInfo
 		var cnsVolumeInfo *cnsvolumeinfov1alpha1.CNSVolumeInfo
+		var volumeSnapshotUID string
 		isStorageQuotaM2FSSEnabled := commonco.ContainerOrchestratorUtility.IsFSSEnabled(ctx,
 			common.StorageQuotaM2)
+		volumeSnapshotName := req.Parameters[common.VolumeSnapshotNameKey]
+		volumeSnapshotNamespace := req.Parameters[common.VolumeSnapshotNamespaceKey]
+		isCSITransactionSupportEnabled := commonco.ContainerOrchestratorUtility.
+			IsFSSEnabled(ctx, common.CSITranSactionSupport)
+		if isCSITransactionSupportEnabled {
+			volumeSnapshotUID, err = commonco.ContainerOrchestratorUtility.
+				GetVolumeSnapshotUID(ctx, volumeSnapshotName, volumeSnapshotNamespace)
+			if err != nil {
+				return nil, logger.LogNewErrorCodef(log, codes.Internal,
+					"failed to retrieve volumesnapshot UID for volumesnapshot: %q in namespace: %q, Error: %+v",
+					volumeSnapshotName, volumeSnapshotNamespace, err)
+			}
+		}
```

Comment:
req.name is nothing but the volumesnapshot UUID, this can be used as snapshot id

Reply:
done. Used req.name as Snapshot UID.